spring-grpc-core/src/main/java/org/springframework/grpc/client/DefaultGrpcChannelFactory.java

```diff
@@ -71,68 +66,41 @@ public void setCredentialsProvider(ChannelCredentialsProvider credentials) {
 	}
 
 	@Override
-	public ManagedChannelBuilder<?> createChannel(String authority) {
-		return this.createChannel(authority, List.of(), false);
-	}
-
-	@Override
-	public ManagedChannelBuilder<?> createChannel(String authority, List<ClientInterceptor> interceptors,
-			boolean mergeWithGlobalInterceptors) {
-		Assert.notNull(interceptors, () -> "interceptors must not be null");
-		return this.builders.computeIfAbsent(authority, path -> {
-			ManagedChannelBuilder<?> builder = newChannelBuilder(this.targets.getTarget(path),
-					this.credentials.getChannelCredentials(path));
-			this.interceptorsConfigurer.configureInterceptors(builder, interceptors, mergeWithGlobalInterceptors);
-			this.customizers.forEach((c) -> c.customize(path, builder));
-			return new DisposableChannelBuilder(authority, builder);
-		});
+	public GrpcChannelBuilder createChannel(String target) {
+		ManagedChannelBuilder<?> builder = newChannelBuilder(this.targets.getTarget(target),
+				this.credentials.getChannelCredentials(target));
+		this.customizers.forEach((c) -> c.customize(target, builder));
+		return new GrpcChannelBuilder(builder, this::buildAndRegisterChannel);
 	}
 
 	/**
 	 * Creates a new {@link ManagedChannelBuilder} instance for the given target path and
 	 * credentials.
 	 * @param path the target path for the channel
 	 * @param creds the credentials for the channel
-	 * @return a new {@link ManagedChannelBuilder} for the given path and credentials
+	 * @return a new builder for the given path and credentials
 	 */
 	protected ManagedChannelBuilder<?> newChannelBuilder(String path, ChannelCredentials creds) {
 		return Grpc.newChannelBuilder(path, creds);
 	}
 
-	private ManagedChannel buildAndRegisterChannel(String channelName, ManagedChannelBuilder<?> channelBuilder) {
-		return this.channels.computeIfAbsent(channelName, (__) -> channelBuilder.build());
```

Comment:
Caching of channels is gone as well (**for now**).

I recommend that we skip caching of the channels for now. Because we hand out the builders for post-handout mutations, we would need to be able to `Function<Builder, CacheKey>` and that is a bumpy road that I would like to holdoff traveling down.

We still handle the lifecycle (shutdown), but users have to cache the channels themselves.

Wdyt?

Reply:
Caching of channels is still there in `this.channels`. I think that makes sense - not sure how we'd handle lifecycle without caching channels.

Reply:
**[Discussed offline]** summary: the channels are "remembered" so we can handle lifecycle but not "cached" in traditional sense from a user perspective (i.e. if a user asks for the same target 2x, they get 2x channels, not 1).